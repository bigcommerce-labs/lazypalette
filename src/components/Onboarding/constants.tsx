// tslint:disable:max-line-length
export const OnboardingTitle = 'Welcome to Store Design!';
export const OnboardingMessage = `We've made significant improvements to the way you customize the look and feel of your store.`;
export const OnboardingAction = 'Start exploring';
// tslint:enable:max-line-length
