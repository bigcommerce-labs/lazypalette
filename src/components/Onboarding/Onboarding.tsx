import * as queryString from 'query-string';
import React, { PureComponent } from 'react';
import { connect } from 'react-redux';

import { setQueryParams, QueryParamsData, UpdateQueryParamsAction } from '../../actions/merchant';
import { State } from '../../reducers/reducers';
import { updateQueryParamsService } from '../../services/queryParam';
import AlertModal from '../Modal/AlertModal/AlertModal';

import { OnboardingAction, OnboardingMessage, OnboardingTitle } from './constants';

interface OnboardingProps {
    queryParams: string;
    setQueryParams(queryData: QueryParamsData): UpdateQueryParamsAction;
}

export class Onboarding extends PureComponent<OnboardingProps, {}> {
    checkShouldShowOnboarding = () => {
        const queryParams = queryString.parse(this.props.queryParams);

        return queryParams.optIn === 'true';
    };

    onboardingComplete = () => {
        const queryParams = updateQueryParamsService({ optIn: null });
        this.props.setQueryParams({ queryParams });
    };

    render() {
        if (this.checkShouldShowOnboarding()) {
            return (
                <AlertModal
                    actionText={OnboardingAction}
                    title={OnboardingTitle}
                    body={OnboardingMessage}
                    primaryAction={this.onboardingComplete}
                />
            );
        } else {
            return null;
        }
    }
}

const mapStateToProps = (state: State) => ({
    queryParams: state.merchant.queryParams,
});

const mapDispatchToProps = {
    setQueryParams,
};

export default connect(mapStateToProps, mapDispatchToProps)(Onboarding);
