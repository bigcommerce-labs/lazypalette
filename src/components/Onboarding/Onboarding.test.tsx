import { shallow } from 'enzyme';
import React from 'react';

import { Onboarding } from './Onboarding';

describe('<Onboarding />', () => {
    const mockSetQueryParams = jest.fn();
    beforeEach(() => {
        mockSetQueryParams.mockReset();
    });

    describe('when rendering', () => {
        describe('when optIn query parameter is "true"', () => {
            it('renders correctly', () => {
                const onboarding = shallow(
                    <Onboarding
                        queryParams={'blah&optIn=true'}
                        setQueryParams={mockSetQueryParams}
                    />);

                expect(onboarding).toMatchSnapshot();
            });
        });

        describe('when optIn query parameter is not "true"', () => {
            it('renders correctly', () => {
                const onboarding = shallow(
                    <Onboarding
                        queryParams={'blah&optIn=blah'}
                        setQueryParams={mockSetQueryParams}
                    />);

                expect(onboarding).toMatchSnapshot();
            });
        });

        describe('when optIn query parameter is not present', () => {
            it('renders correctly', () => {
                const onboarding = shallow(
                    <Onboarding
                        queryParams={'blah'}
                        setQueryParams={mockSetQueryParams}
                    />);

                expect(onboarding).toMatchSnapshot();
            });
        });
    });

    describe('when onboarding is complete', () => {
        it('removed the optIn query parameter', () => {
            const onboarding = shallow(
                <Onboarding
                    queryParams={'?variationId=123&optIn=true'}
                    setQueryParams={mockSetQueryParams}
                />);

            expect(mockSetQueryParams).not.toHaveBeenCalled();
            (onboarding.instance() as Onboarding).onboardingComplete();
            expect(mockSetQueryParams).toHaveBeenCalled();
        });
    });
});
