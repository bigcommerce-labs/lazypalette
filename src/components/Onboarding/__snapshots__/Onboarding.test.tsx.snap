// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Onboarding /> when rendering when optIn query parameter is "true" renders correctly 1`] = `
<AlertModal
  actionText="Start exploring"
  body="We've made significant improvements to the way you customize the look and feel of your store."
  primaryAction={[Function]}
  title="Welcome to Store Design!"
/>
`;

exports[`<Onboarding /> when rendering when optIn query parameter is not "true" renders correctly 1`] = `""`;

exports[`<Onboarding /> when rendering when optIn query parameter is not present renders correctly 1`] = `""`;
